pulser/simulation/simulation.py

```diff
@@ -90,26 +100,60 @@ def __init__(self, sequence: Sequence, sampling_rate: float = 1.0,
         self._times = self._adapt_to_sampling_rate(
             np.arange(self._tot_duration, dtype=np.double)/1000)
         self.evaluation_times = evaluation_times
-        self.config = config if config else SimConfig()
+        self._config = config if config else SimConfig()
+        self._update_noise()
+        self._extract_samples()
+        self._build_basis_and_op_matrices()
+        self._construct_hamiltonian()
         self.initial_state = 'all-ground'
 
     @property
     def config(self) -> SimConfig:
         """Property getter for config."""
         return self._config
 
-    @config.setter
-    def config(self, cfg: SimConfig) -> None:
+    def set_config(self, cfg: SimConfig) -> None:
+        """Sets current config to cfg and updates simulation parameters.
+
+        Args:
+            cfg (SimConfig): New configuration.
+        """
+        if not isinstance(cfg, SimConfig):
+            raise ValueError(f"Object {cfg} is not a valid `SimConfig`")
         self._config = cfg
         self._set_param_from_config()
 
+    def update_config(self, config: SimConfig) -> None:
+        """Updates this SimConfig object with parameters of another one.
+
+        Mostly useful when dealing with multiple noise types in different
+        configurations and wanting to merge these configurations together.
+
+        Args:
+            config (SimConfig): SimConfig to retrieve parameters from.
+        """
+        old_noise_set = set(self.config.noise)
+        new_noise_set = old_noise_set.union(config.noise)
+        diff_noise_set = new_noise_set - old_noise_set
+        param_dict: dict[str, Any] = {}
+        param_dict['noise'] = tuple(new_noise_set)
+        if 'SPAM' in diff_noise_set:
+            param_dict['eta'] = config.eta
+            param_dict['epsilon'] = config.epsilon
+            param_dict['epsilon_prime'] = config.epsilon_prime
+        if 'doppler' in diff_noise_set:
+            param_dict['temperature'] = config.temperature
+        if 'amplitude' in diff_noise_set:
+            param_dict['laser_waist'] = config.laser_waist
+        self.set_config(SimConfig(**param_dict))
```

Comment:
Isn't this just creating a new `SimConfig` with only the differences between the old and the new? Should you start with `param_dict` having the parameters of `old_dict`? (Look at [this](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) for that)